### PR TITLE
support castling methods

### PR DESCRIPTION
**Pull request by ZoeBKramer**

### CODE REVIEW

**Review by BrandonJohnson65 — COMMENTED** [review]
This looks great!!! My only question might be if the horizontal obstructions are accounted for because you can only make a castling move if the spaces in-between the castle and king are empty. I may have missed it somewhere in there but if not, you could just plug the horizontal_obstruction? method into the _valid? methods as a boolean and it should be good to go. This looks awesome.

**Comment by ZoeBKramer** [comment]
> This looks great!!! My only question might be if the horizontal obstructions are accounted for because you can only make a castling move if the spaces in-between the castle and king are empty. I may have missed it somewhere in there but if not, you could just plug the horizontal_obstruction? method into the _valid? methods as a boolean and it should be good to go. This looks awesome.

Oof you're totally right, can't believe I forgot that lol. Good catch! I will add that and send a new commit right away. Thanks Brandon.

File: app/models/king.rb [hunk 1]
```diff
@@ -12,6 +12,77 @@ def valid_move?(x, y)
...
+  end
+
+  def kingside_valid?(x, y)
+    if x_space + 2 == x and
+      move_count == nil and
+      y_space == y 
+      if color = "black"
+        kingside_rook = Piece.find(x_space: 7, y_space: 0, type: "Rook", color: "black", move_count: nil)
+        if kingside_rook != nil and
+          !is_obstructed?(7, 0)
```
Comment (ZoeBKramer) [inline app/models/king.rb:45]: @BrandonJohnson65, thanks again for reminding me about obstructions! Let me know if there's anything else that looks funky.

Comment (BrandonJohnson65) [reply app/models/king.rb:45]: Looks good to me!

**Review by BrandonJohnson65 — APPROVED** [review]

**Pull request merged**